### PR TITLE
Added newsletter in get helper

### PR DESCRIPTION
**Pull request by vershwal**
refs https://github.com/TryGhost/Product/issues/3859

### CODE REVIEW

**Comment by mike182uk** [comment]
@vershwal This doesn't seem to be working when I've tried it in a template:

```
{{#get "newsletters" as |newsletter|}}
  {{#foreach newsletter}}
    {{name}}
  {{/foreach}}
{{/get}}
```

File: ghost/core/core/frontend/helpers/foreach.js [hunk 1]
```diff
@@ -31,6 +31,12 @@ module.exports = function foreach(items, options) {
         if (Object.values(items).length > 0 && checks.isPost(Object.values(items)[0])) {
             visibility = visibility || 'all';
         }
+    } if (_.isArray(items) && items.length > 0 && checks.isNewsletter(items[0])) {
```
Comment (mike182uk) [inline ghost/core/core/frontend/helpers/foreach.js:34]:
```suggestion
    } 
    
    if (_.isArray(items) && items.length > 0 && checks.isNewsletter(items[0])) {
```

Having the `if` on the same line made it look like it was part of the `else if` which it is not

File: ghost/core/core/frontend/services/data/checks.js [hunk 2]
```diff
@@ -3,6 +3,11 @@ function isPost(jsonData) {
         Object.prototype.hasOwnProperty.call(jsonData, 'title') && Object.prototype.hasOwnProperty.call(jsonData, 'slug');
 }
 
+function isNewsletter(jsonData) {
+    return Object.prototype.hasOwnProperty.call(jsonData, 'name') &&
+        Object.prototype.hasOwnProperty.call(jsonData, 'subscribe_on_signup') && Object.prototype.hasOwnProperty.call(jsonData, 'slug');
```
Comment (mike182uk) [inline ghost/core/core/frontend/services/data/checks.js:8]: Maybe we could use less overlapping named fields here? How about `subscribe_on_signup`, `status` and `visibility`?

File: ghost/core/test/unit/frontend/helpers/foreach.test.js [hunk 3]
```diff
@@ -282,6 +290,14 @@ describe('{{#foreach}} helper', function () {
             ]
         };
 
+        const newsletterArrayHashWithVisibility = {
```
Comment (mike182uk) [inline ghost/core/test/unit/frontend/helpers/foreach.test.js:293]: How about we co-locate `newsletterObjectHashWithVisibility` and `newsletterArrayHashWithVisibility` with the test seeming though its unlikely they will be used in other tests?

File: ghost/core/test/unit/frontend/helpers/foreach.test.js [hunk 4]
```diff
@@ -402,6 +402,27 @@ describe('{{#foreach}} helper', function () {
...
+                    third: {name: 'third', visibility: 'paid', slug: 'third', subscribe_on_signup: false}
+                }
+            };
+            const newsletterArrayHashWithVisibility = {
+                newsletters: [
+                    {name: 'first', visibility: 'members', slug: 'first', subscribe_on_signup: true},
+                    {name: 'second', visibility: 'members', slug: 'second', subscribe_on_signup: true},
+                    {name: 'third', visibility: 'paid', slug: 'third', subscribe_on_signup: false}
+                ]
+            };
```
Comment (mike182uk) [inline ghost/core/test/unit/frontend/helpers/foreach.test.js:419]: We don't need `slug` in the test data anymore right?

File: ghost/core/test/unit/frontend/services/data/checks.test.js [hunk 5]
```diff
@@ -26,6 +27,14 @@ describe('Checks', function () {
         checks.isPage({title: 'Test', show_title_and_feature_image: true}).should.eql(true);
     });
 
+    it('isNewsletter', function () {
+        checks.isNewsletter({}).should.eql(false);
+        checks.isNewsletter({name: 'Test'}).should.eql(false);
+        checks.isNewsletter({name: 'Test', slug: 'test'}).should.eql(false);
```
Comment (mike182uk) [inline ghost/core/test/unit/frontend/services/data/checks.test.js:33]: I don't think we need this assertion

**Review by mike182uk — APPROVED** [review]
Couple of minor tweaks @vershwal but good to 🚢

**Comment by vershwal** [comment]
Done 🎉

**Pull request merged**